### DISCUSSION TITLE
id with spaces

### DISCUSSION THREAD

**Discussion by ammar-mu** [free-audio/clap#327] [Q&A] [answered]
Hi, I'm in the last testing phases before releasing a CLAP version of a previously VST 2.4 plugin. So I might be asking a few questions later just to make sure I don't get something wrong.

In clap_plugin_descriptor_t, is it OK to have an id with spaces. Like when instrument name actually contains spaces. Example:

		.id = "com.abc123.My Plugin",
		.name ="My Plugin",

**Comment by defiantnerd** [comment]
It is not being prevented, yet, I would stick with a proper id string without spaces. Replace spaces with underscores if you want?

**Reply by ammar-mu** [reply]
I see. Yeah I'll probably use underscores or even dashes.

**Reply by baconpaul** [reply, marked as answer]
The spec says "any uff-8 string is valid" so in theory your id could be `superAwesome MyThing 💯' and I know some of the plugin developers have made test plugin with names like that to stress test daws. 

But I agree with defiant nerd. For an id keeping in the ascii-no-spaces subset of UTF-8 makes it less likely your plugin is the one which leads to a DAW bug report :)

**Reply by ammar-mu** [reply]
Thanks paul. Makes sense :)
